kf_lib_data_ingest/common/io.py

```diff
@@ -0,0 +1,169 @@
+"""
+Contains file readers (file -> dataframe) for file types that benefit from
+extra help.
+"""
+import io
+import json
+import os
+
+import jsonpickle
+import pandas
+import xlrd
+import yaml
+from pandas.api.types import is_file_like
+
+
+def __ascii_wrap(potentially_binary_buffer):
+    try:
+        return io.TextIOWrapper(potentially_binary_buffer)
+    except Exception:
+        return potentially_binary_buffer
+
+
+def read_excel_df(filepath_or_buffer, **kwargs):
+    """
+    Return contents of an excel spreadsheet as a pandas DataFrame.
+
+    :param filepath_or_buffer: an xls or xlsx spreadsheet file
+    :type filepath_or_buffer: string (path) or file-like object
+    :param **kwargs: See docs for pandas.read_excel
+    :return: The structured contents of the file
+    :rtype: pandas.Dataframe
+    """
+    # Pre-opening the workbook with xlrd lets us suppress noisy warnings
+    # like "WARNING *** OLE2 inconsistency: SSCS size is 0 but SSAT size is
+    # non-zero" by pushing them to /dev/null
+    with open(os.devnull, "w") as devnull:
+        if is_file_like(filepath_or_buffer):
+            wb = xlrd.open_workbook(
+                file_contents=filepath_or_buffer.read(), logfile=devnull
+            )
+        else:
+            wb = xlrd.open_workbook(
+                filename=filepath_or_buffer, logfile=devnull
+            )
+
+    kwargs["engine"] = "xlrd"
+    kwargs["dtype"] = str
+    kwargs["na_filter"] = False
+    return pandas.read_excel(wb, **kwargs)
+
+
+def read_text_df(filepath_or_buffer, **kwargs):
```

Comment:
I liked `read_table_df` better. `read_text_df` makes it sound like you're reading .txt files or something

Reply:
You said you wanted to read txt files tho, lol

Reply:
What about read_delimited_df?

Reply:
> You said you wanted to read txt files though, lol

Ohhh sorry I thought you decided not to add reading of .txt extension.